packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -90,22 +111,33 @@ export class RemoteChannelContext implements IChannelContext {
         this.services.deltaConnection.setConnectionState(connected);
     }
 
+    public rebaseOp(message: ISequencedDocumentMessage, localOpMetadata: unknown) {
+        if (this.isLoaded) {
+            this.services.deltaConnection.rebase(message, localOpMetadata);
+        } else {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            this.pending!.push({ type: "rebase", message, localOpMetadata });
+        }
+    }
+
     public processOp(message: ISequencedDocumentMessage, local: boolean, localOpMetadata: unknown): void {
         this.summarizerNode.invalidate(message.sequenceNumber);
 
         if (this.isLoaded) {
             this.services.deltaConnection.process(message, local, localOpMetadata);
         } else {
-            assert(!local, "Remote channel must not be local when processing op");
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            this.pending!.push(message);
+            this.pending!.push({ type: "process", message, local, localOpMetadata });
         }
     }
 
     public reSubmit(content: any, localOpMetadata: unknown) {
-        assert(this.isLoaded, "Remote channel must be loaded when resubmitting op");
-
-        this.services.deltaConnection.reSubmit(content, localOpMetadata);
+        if (this.isLoaded) {
+            this.services.deltaConnection.reSubmit(content, localOpMetadata);
+        } else {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            this.pending!.push({ type: "resubmit", content, localOpMetadata });
```

Comment:
Curious, when do we hit this branch of code? I think resubmit should be possible only if dds is already loaded.
For all local changes we should force DDS to load (if it's not loaded), there is no benefit of keeping it non-loaded I think